metrics/ci_integrations/test/cli/command/validate_command_test.dart

```diff
@@ -0,0 +1,532 @@
+      () {
+        whenCreateConfiguredParties().thenReturn(configuredParties);
+        whenValidateSourceConfig().thenAnswer(
+          (_) => Future.value(sourceValidationResult),
+        );
+        whenValidateDestinationConfig().thenAnswer(
+          (_) => Future.value(destinationValidationResult),
+        );
+
+        commandStub.run();
+
+        verify(configuredPartiesFactory.create(rawIntegrationConfig)).called(1);
+      },
+    );
+
+    test(
+      ".run() throws a ConfigValidationError if there is an error occurred during the configured parties creation",
+      () {
+        when(
+          rawIntegrationConfigFactory.create(configFilePath),
+        ).thenReturn(rawIntegrationConfig);
+        when(
+          configuredPartiesFactory.create(rawIntegrationConfig),
+        ).thenThrow(Exception());
+
+        expect(() => commandStub.run(), throwsConfigValidationError);
+      },
+    );
+
+    test(
+      ".run() creates the source config validator using the config validator factory and the config from the configured source party",
+      () {
+        whenCreateConfiguredParties().thenReturn(configuredParties);
+        whenValidateSourceConfig().thenAnswer(
+          (_) => Future.value(sourceValidationResult),
+        );
+        whenValidateDestinationConfig().thenAnswer(
+          (_) => Future.value(destinationValidationResult),
+        );
+
+        commandStub.run();
+
+        verify(sourceConfigValidatorFactory.create(sourceConfig)).called(1);
+      },
+    );
+
+    test(
+      ".run() validates the source config using the source config validator",
+      () {
+        whenCreateConfiguredParties().thenReturn(configuredParties);
+        whenValidateSourceConfig().thenAnswer(
+          (_) => Future.value(sourceValidationResult),
+        );
+        whenValidateDestinationConfig().thenAnswer(
+          (_) => Future.value(destinationValidationResult),
+        );
+
+        commandStub.run();
+
+        verify(sourceConfigValidator.validate(sourceConfig)).called(1);
+      },
+    );
+
+    test(
+      ".run() logs a message when validating the source config",
+      () async {
+        whenCreateConfiguredParties().thenReturn(configuredParties);
+        whenValidateSourceConfig().thenAnswer(
+          (_) => Future.value(sourceValidationResult),
+        );
+        whenValidateDestinationConfig().thenAnswer(
+          (_) => Future.value(destinationValidationResult),
+        );
+
+        await commandStub.run();
+
+        verify(writerMock.write(argThat(
+          isConfigValidationMessage(sourceConfig),
+        ))).called(1);
+      },
+    );
+
+    test(
+      ".run() throws a ConfigValidationError if there is an error occurred during the source config validation",
+      () {
+        whenCreateConfiguredParties().thenReturn(configuredParties);
+        whenValidateSourceConfig().thenAnswer(
+          (_) => throw Exception(),
+        );
```

Comment:
```suggestion
        whenValidateSourceConfig().thenThrow(Exception());
```
Please, check all tests and apply this suggestion to all similar situations.